src/usagemetrics/eda_user_metrics_client.py

```diff
@@ -0,0 +1,96 @@
+import json
+import sys
+from http import client
+from urllib.parse import urlparse
+import pandas
+import pandas as pd
+from usagemetrics.analysis_metrics import AnalysisMetrics
+
+
+class EdaUserServiceMetricsClient:
+
+    def __init__(self, url, project_id):
+        self.url = url
+        self.project_id = project_id
+
+    def query_analysis_metrics(self, start_date, end_date):
+        """
+        # Queries the user service and returns a dataframe with a column index on object type and a row index on metric
+        # histogram bucket. The cells contain the number of users in each bucket based on how many of the object type they
+        # own.
+
+        :param start_date:
+        :param end_date:
+        :return:
+        """
+        eda_url_parse_result = urlparse(self.url)
+        if eda_url_parse_result.scheme == 'https':
+            eda_client = client.HTTPSConnection(str(eda_url_parse_result.hostname), port=eda_url_parse_result.port)
+        else:
+            eda_client = client.HTTPConnection(str(eda_url_parse_result.hostname))
+        # Add this header if using an internal dev or qa site. "Cookie": "auth_tkt=xxx"
+        print(f"Hostname {eda_url_parse_result.hostname} Path: {str(eda_url_parse_result.path)}/metrics/user/{self.project_id}/analyses?startDate={start_date.isoformat().split('T')[0]}&endDate={end_date.isoformat().split('T')[0]}")
+        eda_client.request(method="GET",
+                           url=f"{str(eda_url_parse_result.path)}/metrics/user/{self.project_id}/analyses?startDate={start_date.isoformat().split('T')[0]}&endDate={end_date.isoformat().split('T')[0]}",
```

Comment:
Minor nitpick, but it might make reading this easier if this format string was extracted out into a variable and used in the both the print and this `url` param.